src/Ratchet/Server/IoServer.php

```diff
@@ -80,61 +114,61 @@ public function run() {
      * @param \React\Socket\ConnectionInterface $conn
      */
     public function handleConnect($conn) {
-        $conn->decor = new IoConnection($conn);
-        $conn->decor->resourceId = (int)$conn->stream;
+        $io_conn = new IoConnection($conn);
```

Comment:
I'm guessing `$conn->decor` should still be left to avoid a breaking change

Reply:
decor is not a property of \React\Socket\ConnectionInterface so can't be left in because php 8 deprecated dynamic property creation.

Reply:
Then maybe we also make a PR to reactphp/socket to add #[AllowDynamicProperties] to the interface https://github.com/reactphp/socket/blob/3.x/src/ConnectorInterface.php

Reply:
Or we simply use spl storage locally in IoServer to store the decor and retrieve it when needed